File: Practice /Python/binary_search.py
def find_the_element(user, ele):
    right_boundary = len(user) - 1
    left_boundary = 0
    while(right_boundary >= left_boundary):
        middle_element = (left_boundary+(right_boundary-1) // 2)
        if user[middle_element] == ele:
            return middle_element
        if user[middle_element] > ele:
            right_boundary = middle_element - 1

        if user[middle_element] < ele:
            left_boundary = middle_element + 1
    return 0

user_list = []
size = int(input("please enter the size of the array : "))
for i in range(size):
    user_list.append(int(input("element :")))
element = int(input("Element to be searched : "))

a = find_the_element(user_list, element)
if a != 0:
    print("found at " + str(a))
else:
    print("not found")

File: Practice /C++/binary_search.cpp
#include <iostream>

using namespace std;

int binary_search(int a[],int size,int ele){

  int middle_element;
  int l = 0;
  int r = size - 1;
  while(r >= l){
    middle_element = l + (r-l) / 2;

    if(a[middle_element] == ele){
      return middle_element;
    }

    else if(a[middle_element] < ele){
      l = middle_element + 1;
    }
    else{
      r = middle_element - 1;
    }
  }
}

int main(){

  int array[] = {1,3,4,5,6};
  int size_of_the_array = sizeof(array)/sizeof(array[0]);
  int element = 5;
  int found = binary_search(array,size_of_the_array,element);
  cout << found;
  return 0;
}
